baseplate/thrift.go

```diff
@@ -0,0 +1,164 @@
+package baseplate
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/getsentry/raven-go"
+	"github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/batcherror"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type baseplateThriftServer struct {
+	thriftServer *thrift.TSimpleServer
+	config       ServerConfig
+	closers      []io.Closer
+	logger       log.Wrapper
+	secretsStore *secrets.Store
+}
+
+func (bts *baseplateThriftServer) Config() ServerConfig {
+	return bts.config
+}
+func (bts *baseplateThriftServer) Secrets() *secrets.Store {
+	return bts.secretsStore
+}
+
+func (bts *baseplateThriftServer) Serve() error {
+	return bts.thriftServer.Serve()
+}
+
+func (bts *baseplateThriftServer) Close() error {
+	var errors batcherror.BatchError
+	errors.Add(bts.thriftServer.Stop())
+	for _, c := range bts.closers {
+		errors.Add(c.Close())
+	}
+	return errors.Compile()
+}
+
+func initLogger(cfg ServerConfig) log.Wrapper {
+	if cfg.Log.Level == "" {
+		cfg.Log.Level = log.InfoLevel
+	}
+	level := cfg.Log.Level
+	log.InitLogger(level)
+	return log.ZapWrapper(level)
```

Comment:
In most cases this shouldn't be the same level.

The level used in config is the minimal level to actually show in the logs, which in most cases should be info level. The log wrapper we used for other initialization are for logging errors happened inside those packages that cannot be communicated to the caller otherwise (for example, filewatcher has a background goroutine watching the file and try to load it and call parser when its content changed), so that in most cases should be error level.

Reply:
It will be much less confusing if all components of the system log to the same level.

Reply:
by that logic why do we even bother to provide different log levels?